src/main/java/uk/gov/companieshouse/api/accounts/validation/LoanValidator.java

```diff
@@ -0,0 +1,74 @@
+package uk.gov.companieshouse.api.accounts.validation;
+
+import org.springframework.stereotype.Component;
+
+import uk.gov.companieshouse.api.accounts.model.rest.smallfull.notes.loanstodirectors.Loan;
+import uk.gov.companieshouse.api.accounts.model.rest.smallfull.notes.loanstodirectors.LoanBreakdownResource;
+import uk.gov.companieshouse.api.accounts.model.validation.Errors;
+
+@Component
+public class LoanValidator extends BaseValidator {
+
+    private static final String LOANS_PATH = "$.loans";
+
+    private static final String LOANS_DIRECTOR_NAME = LOANS_PATH + ".director_name";
+
+    private static final String LOANS_DESCRIPTION = LOANS_PATH + ".description";
+
+    private static final String LOANS_BREAKDOWN_PATH_ADVANCES_CREDIT_MADE = LOANS_PATH + ".breakdown.advances_credits_made";
+
+    private static final String LOANS_BREAKDOWN_PATH_ADVANCES_CREDIT_REPAID = LOANS_PATH + ".breakdown.advances_credits_repaid";
+
+    private static final String LOANS_BREAKDOWN_PATH_BALANCE_AT_PERIOD_START = LOANS_PATH + ".breakdown.balance_at_period_start";
+
+    private static final String LOANS_BREAKDOWN_PATH_BALANCE_AT_PERIOD_END = LOANS_PATH + ".breakdown.balance_at_period_end";
+
+
+	public Errors validateLoan(Loan loan) {
+
+		Errors errors = new Errors();
+
+		if (loan.getDirectorName() == null) {
```

Comment:
We could enforce these conditions using `@NotNull` annotations on the rest model fields - this would be more succinct, but disguises the logic a bit, so really a matter of preference

Reply:
Is there a preference on generally whether to go with annotations rather than logic?